python/ray/data/datasource/file_based_datasource.py

```diff
@@ -124,6 +124,32 @@ def _get_write_path_for_block(
         return posixpath.join(base_path, suffix)
 
 
+class FileExtensionFilter(PathPartitionFilter):
+    """A file-extension-based path filter, that filters out all files that don't end
+    with the provided extension(s).
+    """
+
+    def __init__(
+        self,
+        file_extensions: Union[str, List[str]],
+        allow_if_no_extension: bool = False,
```

Comment:
Could you add a docstring description of this parameter? I didn't understand this parameter until I read the implementation of `_file_has_extension`